### ISSUE TITLE
Returning 0 in scorer doesn't filter out documents

### ISSUE THREAD

**Issue by jrots** [RedisLabsModules/RediSearch#146]
Hi was just testing about to test a custom scoring function as defined here: 
http://redisearch.io/Extensions/
it states that you can filter out documents when returning 0 as score - 
just wrote a dummy function to test that returns 0 for each doucment -
 is it normal that I still get documents back? 
```
double JaymeTestScorer(RSScoringFunctionCtx *ctx, RSIndexResult *h, RSDocumentMetadata *dmd,
                    double minScore) {

    printf("score for %d: %f\n", h->docId, dmd->score);
    return 0;
}
```
```
redisgoapi01:8443> FT.SEARCH testjrots "@bd:[(19900101 (19960930] -@uid:(1034662|757014)" GEOFILTER l 51.05 3.71 10 km SCORER JAYME_TEST WITHSCORES LIMIT 0 1
```
```
1) (integer) 82
2) "ud:1181088"
3) "0"
4)  1) "uid"
    2) "1181088"
    3) "hb"
    4) "65.53"
    5) "bd"
    6) "19940623"
    7) "l"
    8) "51.0544782 3.7178716"
    9) "fmin"
   10) "20"
   11) "fmax"
   12) "23"
   13) "dl"
   14) "20161024"
```
on the server side : 
```
score for 124: 1.000000
score for 392: 1.000000
score for 512: 1.000000
score for 688: 1.000000
score for 823: 1.000000
score for 1155: 1.000000
``` 
etc
Do I need to pass MINSCORE somehow to the query?

**Comment by dvirsky** [comment]
No, this is correct. The 0 filters out currently only if there are actually documents with score greater than 0. Let me fix that, it should take a few minutes.

**Comment by dvirsky** [comment]
@jrots ok, I've actually added a special value, because a scoring function might legitimately return 0, and we want something stronger - that will not even count that result in the total number of results. 

So I've added a special value, `RS_SCORE_FILTEROUT`, which is actually negative infinity (but that might change later internally). If you return that the result will be filtered out and removed from the totals count. Check out master for this change, I'll add this to the next release. 

Documentation has been updated accordingly.

**Comment by jrots** [comment]
Alright nice! That was blazingly fast:)

**Comment by dvirsky** [comment]
Thanks :) Example:

```c
double myScorer(RSScoringFunctionCtx *ctx, RSIndexResult *h, RSDocumentMetadata *dmd,
                double minScore) {
  return RS_SCORE_FILTEROUT;
}
```

**Comment by dvirsky** [comment]
Fixed in #/eb2a121351a842f7e5e21d77edd15eec4b559f1c

To be release in 0.20.1
